[TM ONLY] Gives security 3 tasers roundstart, adds tasers to cargo orders, and removes the roundstart armory dragnets, however still orderable from cargo.

kuricityy opened the pull request:
## About The Pull Request

With [new tasers](https://github.com/tgstation/tgstation/pull/88347), I want to test out how implimenting tasers in security loadouts would work. As it stands, stimulants, insulated from genetics, freedom implant, and voltaic combat heart, counter the new taser. In order to balance giving security tasers, I took away the roundstart dragnets, and made them only orderable from cargo. If we do fully merge this PR in the future, we'll likely be slashing the snare mode for dragnets entirely, leaving only the beacon. You can also order more tasers from cargo, 2 for 6K. I also immensely nerfed the amount of stamina damage that the taser does, making it so it isnt "DEATH FOR ALL ANTAGS!!!" the tool. It also can only really last 2 stuns before needing a recharge. 
(UPDATED IMAGES DOWN BELOW)

## Why It's Good For The Game
I wanted to do a test, specifically to see how most antagonists would handle the new tasers, and how they would work in todays fighting aspects. I have no intentions of fully merging this, unless the people actually want me to (which I heavily doubt.) I just wanted to consider the possibilities of new tasers. As of now, I believe they drain 2x less stamina than an AI turret, and have a short max range. (Added note: Scarp also deflects tasers and makes the TASERER tasered, which is pretty funny.)

## Changelog
:cl:
add: Security sacrificed their beloved dragnet for their ol' reliable tool of the past, the taser.
/:cl:

kuricityy committed:
TM tasers for sec

norsvenska commented:
did you make these edits on an outdated version of the map, only pushing up after? asking because i dont see the battle rifle on any of the screenshots, and icebox + wawa are both failing due to a conflict.

kuricityy commented:
knowing me I probably did, bare with me here I gotta fix this shit up

kuricityy committed:
Revert "TM tasers for sec"

This reverts commit d5f2afb7bb03e3dffb5edc50c7355190feb4232a.

MrMelbert commented:
Can you even reach the one on Tram?

kuricityy commented:
you cant without climbing onto the little holder thingies that was for a joke but now I had to remake the whole commit so im fixing that as we speak (the hardest part of this is github god I hate github)

kuricityy committed:
TM tasers for sec

kuricityy commented:
Alright, I got the new commit out, everything should be in order, just waiting for the checks to finish. In the meantime, have the updated photos. (Also, the new commit includes Nebula, which I completely forgot last time.)
![image](https://github.com/user-attachments/assets/a0d6ec1f-751a-40b4-8236-30ee44b38398)
Metastation
![image](https://github.com/user-attachments/assets/31c5c212-9991-496c-af93-338f2afa0816)
Deltastation
![image](https://github.com/user-attachments/assets/2e9ca163-a441-47f0-9106-2b20944f4408)
Tramstation
![image](https://github.com/user-attachments/assets/ab7cfdb0-8918-4a2d-90f2-83c16d3dfb03)
Birdshot Station
![image](https://github.com/user-attachments/assets/191b15fd-258e-4e6f-8514-c531e31dc743)
Icebox Station
![image](https://github.com/user-attachments/assets/f69647a0-fcb6-4058-b427-f81debf024c4)
Wawastation
![image](https://github.com/user-attachments/assets/a6cec4b0-145d-48b8-8d06-bedaea061470)
Nebulastation

norsvenska commented:
the window in the middle of tram is sitting on top of an airless floor tile and probably shouldnt be

kuricityy committed:
Fixes the airless floortile in tramstation armory.

I goofed again

kuricityy commented:
Right, now with all of that aired out, I would like to establish what we should remove in place of the taser, if anything. As of now, my only suggestion would be to remove a dragnet, or even both roundstart dragnets, and possibly upping the cost of the dragnets from cargo.

necromanceranne commented:
If you're seriously considering PR'ing (not a test) the new tasers into the game, in any of implementation that you feel is appropriate (as armory gear, upgrades for sec from cargo or as standard issue, whatever you can rationalize), I am going to demand a blood sacrifice for it in the form of a permanent removal of DRAGnets.

One reason for this is that it exorcises a horribly designed piece of shit out of the game for good; crusting up balance for almost a decade and warping the game around it now that it has been discovered and knowledge proliferated. The other is that I don't think there is a healthy state of the game where there exists both, since they're functionally doing a very similar job, but one is just up and up kind of bullshit about it. (It's DRAGnets)

I understand that means that the fancy DRAGnet beacon would have to be removed, but if someone wants to re-implement rapid prisoner transportation as a piece of sec gear (maybe a fulton subtype?), I'll accept that. But I don't want that to stop DRAGnets from being removed.

necromanceranne commented:
DRAGnets salted my crops, broke the banks of my rivers and streams, poisoned the well, and brought a plague upon the village.

kuricityy commented:
I dont really intend on fully merging this, but if I did, I feel like COMPLETELY removing the dragnet is a bit much- especially when instead we could just nerf the shit out of it- and make it not in the armory roundstart. I think, if anything, we just completely slash the snare mode, and leave the dragnet beacon be. The beacon is a really niche mechanic, but we can atleast leave it in, for the chads that DO like to use it- other than the people who just use dragnet for snare. I personally dont use the beacon feature, but I think its a nice addition for warden to have an opportunity to get some more action. (I love my wardens; normalize using dragnet beacons)

kuricityy commented:
You could say im being biased, but keeping the dragnet beacon ability sounds fine to me. Hell, I'll even rename and resprite it so it looks different if you want me to.

necromanceranne commented:
That's a pretty reasonable alternative, though I think the gun would need to be bulky + can't be dual-wielded on top. My lesser concern is that it ostensibly functions as an energy shotgun at extremely close range, and for a normal sized weapon has some of he highest damage output for size. I think it would need to end up being comparable to other armory options for me to be comfortable with its existence.

I....honestly don't think you could get worse than the coder sprite it has right now.

kuricityy committed:
Merge branch 'master' into sec-for-tasers

kuricityy committed:
Merge remote-tracking branch 'upstream/master' into sec-for-tasers

kuricityy commented:
I threw together a showcase video for anyone who didnt look at the actual new taser pr and wanted some more info on how it would work:
https://youtu.be/xLTMBU0FZ1I

Krail commented:
Oh wow, another "Test merge" that **surely** won't be added to actual merge.


What, we forgot about how tasers were cancer of the game back 10~ years ago, be it yog/bee or whatever stations? And most of people from that time don't care that "It actually takes extra couple seconds to be cringe stun".

There is reason why they were replaced with less cancerous disablers... so... why we even considering this? Since when stun combat is priority? Can we actually have "Why is this good for the game" that isn't "Lol we testing".

necromanceranne commented:
Mate, our tasers got completely reworked a few weeks ago, what the fuck are you on about

Steel-Serperior commented:
> Mate, our tasers got completely reworked a few weeks ago, what the fuck are you on about? They even have a video demonstrating how it works.

I still think these reworked tasers are a bit much. Sec getting a ranged option that both slows down and drains stamina just seems too wacky powerful and going to push stun resistance even more into the must have category for antags.

kuricityy commented:
> I still think these reworked tasers are a bit much. Sec getting a ranged option that both slows down and drains stamina just seems too wacky powerful and going to push stun resistance even more into the must have category for antags.

Im also worried about this- thats why I wanted to actually test it until then. Even if it did get merged, we would get rid of dragnet snares, arguably the best tool sec has for slowdown at this point, I am, however curious how stims would work with the taser- I dont know if stims actually outpace the stam damage for the taser- might have to check that out.

sylvia-from-fulp-station commented:
There is a singular real tazer on my in progress map in the corporate showroom also on a fancy table and now i feel sad

Sakamoto4ka commented:
Buffed disablers

Krail commented:
> Mate, our tasers got completely reworked a few weeks ago, what the fuck are you on about? They even have a video demonstrating how it works.

I already stated it's "Cringe but with extra steps"

Can you like explain why this is TEST MERGE if it's clearly been reworked and couldn't be in hell just be another cancer that plagues TG?

Even in downstream, usually if some gun is added, it's added. No test merges, no testing. How come this clearly reworked taser cancer needs a test merge, if it's so trustworthy reworked? Just add it then without TM?

necromanceranne commented:
Okay.

FalloutFalcon commented:
> > Mate, our tasers got completely reworked a few weeks ago, what the fuck are you on about? They even have a video demonstrating how it works.
> 
> I already stated it's "Cringe but with extra steps"
> 
> Can you like explain why this is TEST MERGE if it's clearly been reworked and couldn't be in hell just be another cancer that plagues TG?
> 
> Even in downstream, usually if some gun is added, it's added. No test merges, no testing. How come this clearly reworked taser cancer needs a test merge, if it's so trustworthy reworked? Just add it then without TM?
> 
> Or... holy crap, it might be TM cause the author of dang PR isn't even sure if it's good for the game.... wow crazy, cause tasers were shit cancer for years.
> 
> Edit: I don't care that they take extra couple seconds to shit on you, it's basically just "Taser at home" but I don't think we want "Taser at home". This is coming from a guy who plays on server that has gun cargo of syndicate level on server. Ya lot clearly aren't sure if this is even good addition.
> 
> Why is it singular taser? Cause you think two or more would shit on balance.
> 
> Why is it only TM? Cause you think it will shit on balance.
> 
> **We can keep going, if it was trustworthy weapon and what not, ya wouldn't even need a test merge nor limitation on how many can spawn.**

None of this is useful information.

dalmationer commented:
My proposal for some neat counterplay to this is making it so that you can try to pull the electrodes out at a big stamina hit (likely to put you in stamcrit normally) that is reduced if stimulants or other stam-restoring chems/pump-up are in your system. This helps a little with letting stims still be a get out of jail card and rewards planning.... OFC this is down to melbert, rather than you, because it's about how the tasers actually work.

Otherwise, I really like this idea. I'm interested to see how it plays.

Krail commented:
> > > Mate, our tasers got completely reworked a few weeks ago, what the fuck are you on about? They even have a video demonstrating how it works.
> > 
> > 
> > I already stated it's "Cringe but with extra steps"
> > Can you like explain why this is TEST MERGE if it's clearly been reworked and couldn't be in hell just be another cancer that plagues TG?
> > Even in downstream, usually if some gun is added, it's added. No test merges, no testing. How come this clearly reworked taser cancer needs a test merge, if it's so trustworthy reworked? Just add it then without TM?
> > Or... holy crap, it might be TM cause the author of dang PR isn't even sure if it's good for the game.... wow crazy, cause tasers were shit cancer for years.
> > Edit: I don't care that they take extra couple seconds to shit on you, it's basically just "Taser at home" but I don't think we want "Taser at home". This is coming from a guy who plays on server that has gun cargo of syndicate level on server. Ya lot clearly aren't sure if this is even good addition.
> > Why is it singular taser? Cause you think two or more would shit on balance.
> > Why is it only TM? Cause you think it will shit on balance.
> > **We can keep going, if it was trustworthy weapon and what not, ya wouldn't even need a test merge nor limitation on how many can spawn.**
> 
> None of this is useful information.

None of this is meant to be useful informaton.

We used to have tasers and it fucking sucked for both security and antags, I wan't to hear "Why is it go for the game" not to be just "LOOOOL WE TESTING FAILED MECHANIC AFTER 6 YEARS".


It was dogshit back then, and I bet it will be dogshit right now.

Otherwise, try to explain me why its "SINGULAR" taser, and why is it being tested instead of actually just fully put into the armory/cargo? Why is it ONE taser? Why is it just being TESTED? If ya lot are sure that it's good for the game, why is it being so "CAREFULLY" tested?



Like how come downstreams have more broken weapons that literally shit on antags and everyone else, but even then we can at least see sometype of balance?

dalmationer commented:
Krail, the thing that we're testing here is called the same thing as the old tasers, but it is actually significantly different. I think you may have some ideas of what this will be like based on the old tasers, rather than this, which is a new and different thing with the same name.

Krail commented:
> Krail, the thing that we're testing here is called the same thing as the old tasers, but it is actually significantly different. I think you may have some ideas of what this will be like based on the old tasers, rather than this, which is a new and different thing with the same name.


The whole facade of "Test merge" is already dead by most of TG players, and I'm not even tg player.

What happens is "Test merge" turns into actualy PR., It happend before and I don't trust that it won't happen again.

Frankly I just don't want TG to make mistakes that are literally fucking 10 years old, yeah sure sounds cringe. But seeing fucking server repeating mistakes from literally, and I mean literally from 10 years ago is something else. What, time suddenly heals stuff?

Tasers are ass, they been ass before and they will be ass, the only difference is that ppl hit by it will have to powergame more and abuse bugs, that's all.

dalmationer commented:
> > Krail, the thing that we're testing here is called the same thing as the old tasers, but it is actually significantly different. I think you may have some ideas of what this will be like based on the old tasers, rather than this, which is a new and different thing with the same name.
> 
> The whole facade of "Test merge" is already dead by most of TG players, and I'm not even tg player.
> 
> What happens is "Test merge" turns into actualy PR., It happend before and I don't trust that it won't happen again.
> 
> Frankly I just don't want TG to make mistakes that are literally fucking 10 years old, yeah sure sounds cringe. But seeing fucking server repeating mistakes from literally, and I mean literally from 10 years ago is something else. What, time suddenly heals stuff?
> 
> Tasers are ass, they been ass before and they will be ass, the only difference is that ppl hit by it will have to powergame more and abuse bugs, that's all.

I'm not sure why you quoted me in this, I didn't mention anything about test merges.
Test merges also don't reach downstreams, so you should be good, anyway?

Iamgoofball commented:
>Tasers are ass, they been ass before and they will be ass, the only difference is that ppl hit by it will have to powergame more and abuse bugs, that's all.

we literally reworked tasers from scratch, they're a completely different item now

Krail commented:
> > Tasers are ass, they been ass before and they will be ass, the only difference is that ppl hit by it will have to powergame more and abuse bugs, that's all.
> 
> we literally reworked tasers from scratch, they're a completely different item now

So... why we are doing "SINGLE" tasers PR?

Goof ya been in TG for long enough to know that possability for absolute bs due to PR is not 0 percent chance.


Like why is it draft? Why is it "Singular taser"? Has anyone even tested it? No. Would shit hit the fan if ya got 5 officers with it?



Ya lot can keep repeating mistakes, all good with me.

DrDiasyl commented:
@Krail 
<details>

![rage](https://media1.tenor.com/m/Y_sIGs9yBcYAAAAd/rage.gif)

_"THE THING THEY ARE HIDING FROM YOU ALL IS THAT ALL PRs ARE CREATED TO BE BAD TO THE GAME. HAVE YOU EVER NOTICED THAT ALL THE 'BAD' PRs ARE FIRST TEST MERGED? THAT IS BECAUSE THEY WANT TO MAKE SURE THEY ARE AS MISERABLE AS POSSIBLE. WHILE MID PRs GET MERGED RIGHT AWAY AND ACTUALLY GOOD PRs NEVER GET PULL REQUESTED AS ALL SMART PEOPLE ALREADY REALIZED THAT BIG CODE WON'T ALLOW THEM TO POST THEM, AND SO THEY REFUGEED TO SOME FURRY SERVER. DON'T BELIEVE ME? BEHEADING - TMd FIRST, BATON ARMOR - TMd FIRST! **THIS IS ONLY THE START, WAKE UP SPESSMAN AS THEY WANT TO REPLACE YOU!! CODE WAR IS NOW!!**"_

</details>

Krail commented:
> @Krail
> 
> ![rage](https://camo.githubusercontent.com/c3436a301e68e777ec0993803e775bed9e8a9f7218445f102dae9629729c3fee/68747470733a2f2f6d65646961312e74656e6f722e636f6d2f6d2f595f73494773397942635941414141642f726167652e676966) [ ![rage](https://camo.githubusercontent.com/c3436a301e68e777ec0993803e775bed9e8a9f7218445f102dae9629729c3fee/68747470733a2f2f6d65646961312e74656e6f722e636f6d2f6d2f595f73494773397942635941414141642f726167652e676966) ](https://camo.githubusercontent.com/c3436a301e68e777ec0993803e775bed9e8a9f7218445f102dae9629729c3fee/68747470733a2f2f6d65646961312e74656e6f722e636f6d2f6d2f595f73494773397942635941414141642f726167652e676966) [ ](https://camo.githubusercontent.com/c3436a301e68e777ec0993803e775bed9e8a9f7218445f102dae9629729c3fee/68747470733a2f2f6d65646961312e74656e6f722e636f6d2f6d2f595f73494773397942635941414141642f726167652e676966)
> 
> _"THE THING THEY ARE HIDING FROM YOU ALL IS THAT ALL PRs ARE CREATED TO BE BAD TO THE GAME. HAVE YOU EVER NOTICED THAT ALL THE 'BAD' PRs ARE FIRST TEST MERGED? THAT IS BECAUSE THEY WANT TO MAKE SURE THEY ARE AS MISERABLE AS POSSIBLE. WHILE MID PRs GET MERGED RIGHT AWAY AND ACTUALLY GOOD PRs NEVER GET PULL REQUESTED AS ALL SMART PEOPLE ALREADY REALIZED THAT BIG CODE WON'T ALLOW THEM TO POST THEM, AND SO THEY REFUGEED TO SOME FURRY SERVER. DON'T BELIEVE ME? BEHEADING - TMd FIRST, BATON ARMOR - TMd FIRST! **THIS IS ONLY THE START, WAKE UP SPESSMAN AS THEY WANT TO REPLACE YOU!! CODE WAR IS NOW!!**"_

Fun one, but again, I don't care what happens to TG terry or bee or yog for all I care, realistically you lot are repeating mistakes from 2013

and ya can ask admin to check my profile, it will match it, at one point maybe we don't fuck around and find out for once?

DrDiasyl commented:
Actually would be great to utilize the Discord discussion feature on this PR

Krail commented:
> if you don't care about /tg/ why are you commenting on /tg/ prs

Cause you goof, ya know it effects downstream.

And I do want ss13 to not die, even if it's like CM etc.


I want ss13 to actually be good, sure it has bumps and what not... but really bringing tasers back after fucking 10 years is something else.

FalloutFalcon commented:
> but really bringing tasers back after fucking 10 years is something else.

its almost like no one said anything about actually bringing them back proper.

DrDiasyl commented:
> Like why it takes people like Livrah to have shit fixed, that is clearly broken for years?

Because it doesn't? It takes an issue report for things to be fixed, and it just so happens that Livrah finds a lot of broken things. And due his large audience, if not him but at least someone will make a issue report.

Krail commented:
> > but really bringing tasers back after fucking 10 years is something else.
> 
> its almost like no one said anything about actually bringing them back proper.

What we never had a PR that was "Purely Test merge" pass as PR? Look at some merged stuff, ya will be shocked.

"We test stuff, no intetions to pass it" > next moth we have PR that juss straight up passes the "testing stuff"

kuricityy commented:
> Like why is it draft? Why is it "Singular taser"? Has anyone even tested it? No. Would shit hit the fan if ya got 5 officers with it?


1, it’s a draft because i’m not sure about its current state. 2, it’s a singular taser because we’re doing a magic thing called a test to see if the taser fits into the sec arsenal now that they aren’t complete cancer, I’m not going to give them  more than one for a test, soz.  3, there is actively a video of me testing it and showing off how it’s different from the old cancerous stun orbs they used to be. 4, you can’t even tase 5 people without recharging twice. (not to mention the smart officers will bring a ballistic shield which directly counters the electrodes)
can I go back to sleep yet

FalloutFalcon commented:
> can I go back to sleep yet

its ok soldier, get your nap, this is the least useful conversation that has ever been held, ever.

Jacquerel commented:
> Actually would be great to utilize the Discord discussion feature on this PR

I'll be honest I personally don't like that feature all that much but I'll consider it when it isn't four people arguing with a single guy. Right now the PR only seems active because one person is very confused and maybe needs to go to bed, not because it's particularly widely controversial.

necromanceranne commented:
From personal experience I've found the feature to be kind of restrictive and actually hampered proper discussion. While we've had one guy on off pretty hard here, I almost guarantee that the best feedback and discussion you're gonna get is here, on the PR.

kuricityy commented:
Right- I think I've decided, for the duration of the actual test merge, I will be replacing the roundstart armory dragnets, with the taser. It'll still be able to be ordered from cargo- just not in the armory roundstart. If this PR does end up getting merged, somehow- I'll probably just end up doing an entire rework of the dragnet- likely completely slashing the snare feature.

kuricityy commented:
Once I get home to make the changes, I'll probably mark it as ready for review.

LemonInTheDark commented:
I don't understand why we're testing this if it's never gonna go anywhere. @MrMelbert will pay for his crimes

kuricityy commented:
> I don't understand why we're testing this if it's never gonna go anywhere. @MrMelbert will pay for his crimes

I’m taking the same way or view Anne took for her baton “nerf.” If the playerbase, excluding sec (we all know they’ll love it) wants me to make it a real PR, yeah sure, fuck it. I have no intention of merging this with my own perspective, I rather want to see the playerbases perspective.

Iamgoofball commented:
Whoa whoa whoa, why's the DRAGNet removed from the armory? That thing is cool.

necromanceranne commented:
I would really appreciate the new tasers being put to the test so I know whether or not they are in a state to reintroduce to the game or not, or whether or it will negatively warp the game around them.

kuricityy commented:
> Whoa whoa whoa, why's the DRAGNet removed from the armory? That thing is cool.

I agree it’s cool, but not only am I not sure of the state of having both tasers AND dragnets. Also the dragnet is definitely op, and I think anyone else would agree. But it’s really not hard to replace otherwise- just one departmental or private order and you’ve got more dragnets than they used to roundstart. If this does become a real PR, i intend on keeping the dragnet in, but just slashing the snare feature. Dragnet beacon my beloved.

terracat474 commented:
wasn't the whole point of taser removal is "erm taser too op we give you melee nerfed stun"

necromanceranne commented:
That was a version of tasers which was just a projectile that stunned on contact. This is an entirely different kind of taser. They're pretty much nothing alike. So there are some unknowns about them that it'd be nice to see play out.

terracat474 commented:
> That was a version of tasers which was just a projectile that stunned on contact. This is an entirely different kind of taser. They're pretty much nothing alike. So there are some unknowns about them that it'd be nice to see play out.

and what's the difference between disabler and taser? why's there a need for another ranged stun gun?

necromanceranne commented:
A disabler functions as most any beam energy projectile does, though it deals stamina damage. 

The taser works in an entirely unique way, creating a tether that causes the damage. There isn't actually anything else in the codebase that works like it.

What place do they have? Well besides the general range differences (the taser only has a range of 5 tiles), that's maybe what we can figure out with testing. Is it too much, is it unnecessary? We don't really know without trying.

DrAmazing343 commented:
Can Freedom Implants break the taser tether like it could the snare bolas? If so, I’m actually a much bigger fan of this than my knee jerk reaction told me I’d be.

necromanceranne commented:
> Can Freedom Implants break the taser tether like it could the snare bolas? If so, I’m actually a much bigger fan of this than my knee jerk reaction told me I’d be.

No, it doesn't, and that's exactly the kind of thing that we need to look at.

kuricityy commented:
> Can Freedom Implants break the taser tether like it could the snare bolas? If so, I’m actually a much bigger fan of this than my knee jerk reaction told me I’d be.

No they aren't. _yet._

DrAmazing343 commented:
Certainly a good idea to make anything that would break the original bola such as Freedom Implants, Lock Heretic’s escape ability, and Ling’s cuff-melting thing do the trick. Definitely balance passes for all of them but ultimately I’d be quite happy if the counterplay is more than just “walk away” because I assume secoffs can also just walk up and baton you while you’re shakin and shivering and screaming from the taser.

otherwise, hell yeah.

Loafin34 commented:
I would say the closest thing to current taser would be  jawed hook fishing rod. also please let us snip the electrodes with wirecutters if we have insuls on, that'd be quite hilarious

btcbuster commented:
why are trying to implement baton nerfs because stun meta too strong and also stronger ranged stuns because stun meta not good enough

kuricityy committed:
Makes stimulants counteract the taser + makes the freedom implant work on the electrode.

also makes genetics insulated mutation taserproof (voltaic heart too) I 100% intend on nerfing this or taking it out if we do actually PR tasers

kuricityy committed:
Merge branch 'sec-for-tasers' of https://github.com/kuricityy/mapping-shit into sec-for-tasers

kuricityy committed:
Removes dragnet from armory, and replaces it with 3 tasers. Also adds taser crates to the cargo shuttle.

You can still buy dragnets from cargo if you wish

LemonInTheDark commented:
> I would really appreciate the new tasers being put to the test so I know whether or not they are in a state to reintroduce to the game or not, or whether or it will negatively warp the game around them.

works for me, go off

kuricityy committed:
Merge branch 'master' into sec-for-tasers

necromanceranne commented on code/modules/projectiles/projectile/energy/stun.dm:
```diff
@@ -184,6 +184,12 @@
+		if(HAS_TRAIT(owner, TRAIT_BATON_RESISTANCE)) // If you have baton resistance while being tased, significantly decreases the stamina damage.
+			to_chat(owner, span_notice("You feel a slight shock, and attempt to shrug it off."))
+			stamina_per_second = 15
+		if(HAS_TRAIT(owner, TRAIT_SHOCKIMMUNE)) // genetics mutation insulated protects from taser shock, as well as voltaic heart
```
We may want to discuss this a bit further, since ``TRAIT_SHOCKIMMUNE`` is possible to acquire in genetics. But let's leave it for another time.

kuricityy replied:
Im aware, I was convinced by the hooligans in dev discussion to add it. I dont intend on keeping it in if we make this a real PR, though,

necromanceranne replied:
The other thing I'm thinking about is that, since the other effect of the taser is a slowdown, that might not make sense if this isn't doing any stamina damage, so maybe this would work as an early return situation similar to the ``TRAIT_PIERCEIMMUNE`` on line 165. Again, something to hash out later.

necromanceranne reviewed (changes requested):
Other than the copy pasta mistake, this is probably ready for testing as far as I can tell. Once thats sorted, we can go from there.

kuricityy committed:
fixes my silly little copy pasta nyoops

kuricityy committed:
Merge branch 'sec-for-tasers' of https://github.com/kuricityy/mapping-shit into sec-for-tasers

kuricityy commented on code/modules/cargo/packs/security.dm:
```diff
@@ -174,13 +174,20 @@
 	discountable = SUPPLY_PACK_RARE_DISCOUNTABLE
 
 /// Armory packs
+/datum/supply_pac/security/armory/tasers
```
fixed in new commit

kuricityy commented on code/modules/projectiles/projectile/energy/stun.dm:
```diff
@@ -93,7 +93,7 @@
 	datum/fired_from,
 	atom/movable/firer,
-	tase_stamina = 80,
+	tase_stamina = 20,
```
after testing in game, the tase_stamina value actually has a very LITTLE impact on the time it takes to stamcrit. Couldnt figure out why, though.

necromanceranne replied:
Ah I see. It's passing the value from the projectile itself to override this one. You'll need to the ``tase_stamina`` var on the projectile.

Jacquerel commented:
> Certainly a good idea to make anything that would break the original bola such as Freedom Implants, Lock Heretic’s escape ability, and Ling’s cuff-melting thing do the trick. Definitely balance passes for all of them but ultimately I’d be quite happy if the counterplay is more than just “walk away” because I assume secoffs can also just walk up and baton you while you’re shakin and shivering and screaming from the taser.
> 
> otherwise, hell yeah.

This would also be worth doing in a separate PR regardless of the reception of any possible test, given that tasers _do_ still exist in the AI core.

optimumtact commented:
awful idea

kuricityy committed:
makes tasers less painful for the general crew

so people dont go down in 3~ seconds

kuricityy commented:
oh i pressed the wrong button its so peak

kuricityy commented:
hold on I probably did something really bad I got this

kuricityy committed:
Merge remote-tracking branch 'upstream/master' into sec-for-tasers

kuricityy commented:
with the help of my favorite professional dumbass I have fixed this


thank vect0r that I didnt have to redo everything

Vect0r2 commented:
> with the help of my favorite professional dumbass I have fixed this
> 
> thank vect0r that I didnt have to redo everything

yeah it would be annoying if you did wouldn't it.

kuricityy committed:
Merge branch 'master' into sec-for-tasers

kuricityy committed:
Merge branch 'master' into sec-for-tasers

necromanceranne commented:
By the by if this PR does close, it can still be testmerged

kuricityy commented:
FUCK OFF STALEBOT!!!

Lorwp commented:
I can testmerge this when conflicts are resolved

LemonInTheDark commented:
@necromanceranne did we get anything from this? you satisfied and all?

necromanceranne commented:
> @necromanceranne did we get anything from this? you satisfied and all?

We never had the pop to really give it a shot to test. So, honestly, maybe we let melbert cook and see how we go.
